Development.md

```diff
@@ -0,0 +1,194 @@
+
+```sh
+$ ./scripts/generate-dev-bundle.sh
+```
+
+This will generate a new tarball (`dev_bundle_<Platform>_<arch>_<version>.tar.gz`) in the root of the checkout.  Assuming you bumped the `BUNDLE_VERSION`, the new version will be extracted automatically when you run `./meteor`.  If you are rebuilding the same version (or didn't bump the version number), you should delete the existing `dev_bundle` directory to ensure the new tarball is extracted when you run `./meteor`.
+
+## Additional documentation
+
+The Meteor core is best documented within the code itself, however, many components also have a `README.md` in their respective directories.
+
+Some compartmentalized portions of Meteor are broken into packages ([see a list of packages](packages/)) and they almost all have a `README.md` within their directory.  For example, [`ddp`](packages/ddp/README.md), [`ecmascript`](packages/ecmascript/README.md) and [`tinytest`](packages/tinytest/README.md).
+
+For the rest, try looking nearby for a `README.md`.  For example, [`isobuild`](tools/isobuild/README.md) or [`cordova`](tools/cordova/README.md).
+
+## Tests
+
+### Test against the local meteor copy
+
+When running any of tests, be sure run them against the checked-out copy of Meteor instead of
+the globally-installed version.  This means ensuring that the command is `path-to-meteor-checkout/meteor` and not just `meteor`.
+
+This is important so that tests are run against the version in development and not the stable (installed) Meteor release.
+
+### Running tests on Meteor core
+
+When you are working with code in the core Meteor packages, you will want to make sure you run the
+full test-suite (including the tests you added) to ensure you haven't broken anything in Meteor. The
+`test-packages` command will do just that for you:
+
+    ./meteor test-packages
+
+Exactly in the same way that [`test-packages` works in standalone Meteor apps](https://guide.meteor.com/writing-atmosphere-packages.html#testing), the `test-packages` command will start up a Meteor app with [TinyTest](./packages/tinytest/README.md).  To view the results, just connect to `http://localhost:3000` (or your specified port) and view the results.
+
+Specific portions of package tests can be run by passing a `<package name>` or `<package path>` to the `test-packages` command. For example, to run `mongo` tests, it's possible to run:
+
+    ./meteor test-packages mongo
+
+### Running Meteor Tool self-tests
+
+While TinyTest and the `test-packages` command can be used to test internal Meteor packages, they cannot be used to test the Meteor Tool itself. The Meteor Tool is a node app that uses a home-grown "self test" system.
+
+For even more details on how to run Meteor Tool "self tests", please refer to the [Testing section of the Meteor Tool README](https://github.com/meteor/meteor/blob/master/tools/README.md#testing).
```

Comment:
It might makes sense to move this sentence to the end of the "Running Meteor Tool self-tests" section. It seems a bit misplaced here, as we just started talking about `self-test` then immediately say "for even more details ...", but we haven't really given any details yet. If placed at the end of this section, then we would have.